src/inspector_profiler.h

```diff
@@ -142,6 +145,63 @@ class V8HeapProfilerConnection : public V8ProfilerConnection {
   bool ending_ = false;
 };
 
+class ProfileWatchdog : public BaseObject {
+ public:
+  enum class ProfileWatchdogState { kInitialized, kRunning, kClosing, kClosed };
+  ProfileWatchdog(Environment* env, v8::Local<v8::Object> object);
+  ~ProfileWatchdog() override;
+  static v8::Local<v8::FunctionTemplate> GetConstructorTemplate(
+      Environment* env);
+  static void Start(const v8::FunctionCallbackInfo<v8::Value>& args);
+  static void Stop(const v8::FunctionCallbackInfo<v8::Value>& args);
+  template <typename Fn>
+  void AddTask(Fn&& cb,
+               CallbackFlags::Flags flags = CallbackFlags::Flags::kRefed);
+  void HandleTasks();
+  void Start();
+  void Stop();
+  void SetTimeout();
+  virtual bool TimeoutHandler() = 0;
+  void set_state(ProfileWatchdogState value) {
+    state_.store(value, std::memory_order_relaxed);
+  }
+  ProfileWatchdogState state() {
+    return state_.load(std::memory_order_relaxed);
+  }
+
+ protected:
+  uint64_t interval_;
```

Comment:
Why `protected`?

Reply:
Because this field will be accessed by subclass. Or add a `interval` method ?

Reply:
Oh, I missed that. Never mind then, `protected` is fine.